### PR TITLE
[monitor query] move logs to new azlogs module

### PR DESCRIPTION
**Pull request by gracewilcox**
Moving the logs functionality to a new module. Part of a wider Monitor restructuring.

Part of https://github.com/Azure/azure-sdk-for-go/issues/22377

### CODE REVIEW

**Review by chlowell — APPROVED** [review]

File: sdk/monitor/query/azlogs/cloud_config.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azlogs
+
+import "github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+
+// Cloud Service Names for Monitor Query Logs and Metrics, used to identify the respective cloud.ServiceConfiguration
+const (
+	ServiceNameLogs cloud.ServiceName = "query/azlogs"
+)
```
[the remark this answers is not in the document]
Comment (gracewilcox) [reply sdk/monitor/query/azlogs/cloud_config.go]: fixed in e23b441

File: sdk/monitor/query/azlogs/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	"github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azlogs"
+	"github.com/stretchr/testify/require"
+)
+
+const recordingDirectory = "sdk/monitor/query/azlogs/testdata"
+const fakeWorkspaceID = "32d1e136-gg81-4b0a-b647-260cdc471f68"
+const fakeWorkspaceID2 = "asdjkfj8k20-gg81-4b0a-9fu2-260c09fn1f68"
+const fakeResourceURI = "/subscriptions/faa080af-c1d8-40ad-9cce-e1a451va7b87/resourceGroups/rg-example/providers/Microsoft.AppConfiguration/configurationStores/example"
+const fakeSubscrtiptionID = "faa080af-c1d8-40ad-9cce-e1a451va7b87"
+const fakeRegion = "westus2"
```
[the remark this answers is not in the document]
Comment (gracewilcox) [reply sdk/monitor/query/azlogs/utils_test.go]: fixed in e23b441

**Comment by gracewilcox** [comment]
@microsoft-github-policy-service rerun

**Pull request merged**